### ISSUE TITLE
springmvc codegen not creating @apiresponse annotations with 'response' field when multiple responses

### ISSUE THREAD

**Issue by scottball1968** [swagger-api/swagger-codegen#2374]
I have an api that has paths that return multiple possible responses that contain objects.  Unlike issue #1877 these responses all return the same object.  Generation code with springmvc codegen is not creating the @apiresponse annotations with the 'response' field completed for the return object.  It does complete the @apioperation annotation with a response field completed.

Swagger-UI and Springfox then do not create api-docs for response models for the non-default response; it does list all the specified responses.

Generating code with the jaxrs jersey language destination does create the expected annotations with response fields set properly.  

Here is a section of my swagger spec of my api,
```
swagger: '2.0'

info:
  version: "1.1 WIP"
  title: Auth
  description: 'Sapphire Authentication & Authorization API (Adding Patient User Groups)'

schemes:
  - https

securityDefinitions:
  token:
    type:  apiKey
    name: auth_token
    in: header
  
paths:
  /authentication/login:
    post:
      tags:
        - authentication
      operationId: login
      description: Authenticates and establishes a session
      consumes:
        - application/json
      produces:
        - application/json
      parameters:
        - name: Credentials
          in: body
          description: credentials used to authenticate user to sapphire
          required: true
          schema:
            $ref: "#/definitions/Credentials"
      responses:
        500:
          description: unexpected error, response code set 0
          schema:
            $ref: "#/definitions/LoginResponse"
        401:
          description: unauthorized, response code 2(account locked), 3(invalid credentials), 4(password expired)
        200:
          description: Successful Login, response code set 1
          schema:
            $ref: "#/definitions/LoginResponse"
definitions:
  Credentials:
    type: object
    required:
      - username
      - password
      - applicationId
    properties:
      username:
        type: string
        example: sapphireclin
      password:
        type: string
      applicationId:
        type: string
        description: The application to which the user is authenticating
  LoginResponse:
    type: object
    required:
    - responseCode
    - token
    - userPermissions
    - userLogin
    - userFirstName
    - userLastName
    - userEmailAddress
    - userTopOrgId
    - passwordMinLength
    properties:
      responseCode:
        type: integer
        format: int32
        description: 1 = Success, 2 = Invalid Credentials, 3 = Password Expired
      token:
        $ref: "#/definitions/Token"
      userPermissions:
        type: array
        items:
          type: string
      userLogin:
        type: string
      userFirstName:
        type: string
      userLastName:
        type: string
      userEmailAddress:
        type: string
      userTopOrgId:
        type: string
      passwordMinLength:
        type: integer
        format: int32
```

And the generated code with spring-mvc language specified:

```
package sample;

import sample.model.*;

import sample.model.LoginResponse;
import sample.model.Credentials;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import io.swagger.annotations.ApiParam;
import io.swagger.annotations.ApiResponse;
import io.swagger.annotations.ApiResponses;
import io.swagger.annotations.Authorization;
import io.swagger.annotations.AuthorizationScope;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

import static org.springframework.http.MediaType.*;

@Controller
@RequestMapping(value = "/authentication", produces = {APPLICATION_JSON_VALUE})
@Api(value = "/authentication", description = "the authentication API")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.SpringMVCServerCodegen", date = "2016-03-13T09:49:34.839-04:00")
public class AuthenticationApi {
  

  @ApiOperation(value = "", notes = "Authenticates and establishes a session", response = LoginResponse.class)
  @ApiResponses(value = { 
    @ApiResponse(code = 200, message = "Successful Login, response code set 1"),
    @ApiResponse(code = 401, message = "unauthorized, response code 2(account locked), 3(invalid credentials), 4(password expired)"),
    @ApiResponse(code = 500, message = "unexpected error, response code set 0") })
  @RequestMapping(value = "/login", 
    produces = { "application/json" }, 
    consumes = { "application/json" },
    method = RequestMethod.POST)
  public ResponseEntity<LoginResponse> login(

@ApiParam(value = "credentials used to authenticate user to sapphire" ,required=true ) @RequestBody Credentials credentials
)
      throws NotFoundException {
      // do some magic!
      return new ResponseEntity<LoginResponse>(HttpStatus.OK);
  }
}
```

And finally with jax-rs jersey language I do get the annotations,

```
package io.swagger.api;

import io.swagger.model.*;
import io.swagger.api.AuthenticationApiService;
import io.swagger.api.factories.AuthenticationApiServiceFactory;

import io.swagger.annotations.ApiParam;

import com.sun.jersey.multipart.FormDataParam;

import io.swagger.model.LoginResponse;
import io.swagger.model.Credentials;

import java.util.List;
import io.swagger.api.NotFoundException;

import java.io.InputStream;

import com.sun.jersey.core.header.FormDataContentDisposition;
import com.sun.jersey.multipart.FormDataParam;

import javax.ws.rs.core.Context;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.SecurityContext;
import javax.ws.rs.*;

@Path("/authentication")


@io.swagger.annotations.Api(description = "the authentication API")
@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaJerseyServerCodegen", date = "2016-03-13T19:17:27.663Z")
public class AuthenticationApi  {
   private final AuthenticationApiService delegate = AuthenticationApiServiceFactory.getAuthenticationApi();

    @POST
    @Path("/login")
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @io.swagger.annotations.ApiOperation(value = "", notes = "Authenticates and establishes a session", response = LoginResponse.class, tags={ "authentication",  })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "Successful Login, response code set 1", response = LoginResponse.class),
        
        @io.swagger.annotations.ApiResponse(code = 401, message = "unauthorized, response code 2(account locked), 3(invalid credentials), 4(password expired)", response = LoginResponse.class),
        
        @io.swagger.annotations.ApiResponse(code = 500, message = "unexpected error, response code set 0", response = LoginResponse.class) })

    public Response login(
@ApiParam(value = "credentials used to authenticate user to sapphire" ,required=true) Credentials credentials,@Context SecurityContext securityContext)
    throws NotFoundException {
        return delegate.login(credentials,securityContext);
    }
}
```

Any ideas other than hand coding the annotations?
thanks

**Comment by wing328** [comment]
@scottball1968 this should be addressed by the PR #2603 submitted by @kolyjjj. Please take a look.

**Comment by wing328** [comment]
PR merged into master. Thanks for the contribution from @kolyjjj
